packages/components/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,116 @@
+import type { SystemProp, Theme } from "@xstyled/styled-components";
+import React from "react";
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { Box } from "../../primitives/Box";
+import { Icon } from "../Icon";
+
+type IconNames = keyof typeof HeroOutlineIcons;
+
+type BadgeSizeOptions = "large" | "small";
+type BadgeVariantOptions = "blue" | "gray" | "green" | "yellow";
+
+export interface BadgeProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, "color"> {
+  /** The contents of the badge. Can be text. */
+  children: NonNullable<React.ReactNode>;
+  /** Icon to be added on the left of the content. */
+  icon?: IconNames;
+  /** Sets the size of the badge. */
+  size?: BadgeSizeOptions;
+  /** Sets the variant of the badge. */
+  variant?: BadgeVariantOptions;
```

Comment:
I think that would be better to call it `color` as we talked on slack.

Reply:
Yeah. I agree. We can save `variant` if we ever add larger badge sizes.